cls/SourceControl/Git/Utils.cls

```diff
@@ -1604,16 +1532,64 @@ ClassMethod RunGitCommandWithInput(command As %String, inFile As %String = "", O
 
     set newArgs($increment(newArgs)) = command
 
+    // defining variables for if statement use later
+    set syncIris = 0
+    set diffBase = ""
+    set diffCompare = ""
+    if (command = "checkout") || (command = "merge") || (command = "rebase") || (command = "pull"){
+        set syncIris = 1
+        set diffCompare = args(args)
+    }
+    
+    
     for i=1:1:$get(args) {
         if ($data(args(i))) {
             set newArgs($increment(newArgs)) = args(i)
+            if (args(i) = "checkout") || (args(i) = "merge") || (args(i) = "rebase") || (args(i) = "pull"){
+                set syncIris = 1
+                set diffCompare = args(i + 1)
+                
+                if args = (i + 2) {
+                    set diffBase = args(i + 2)
+                }
+            }
+        }
+    }
+
+    if (diffCompare = "--no-commit") || (diffCompare = "--abort") {
+        set syncIris = 0
+    }
+
+    if syncIris {
+        if diffBase = "" {
+            set diffBase = ..GetCurrentBranch()
+        }
+        do ..RunGitCommand("fetch", .errorStream, .outputStream)
+        kill errorStream, outputStream
+        do ##class(SourceControl.Git.Utils).RunGitCommandWithInput("diff",,.errorStream,.outputStream, diffBase_".."_diffCompare, "--name-status")
```

Comment:
Pull is not working for me to load changed files into IRIS because the local branch name and the remote branch name are the same. So the command here is "git diff main..main" which detects no differences. I think we need some more logic to determine that the diffCompare is "origin/main" in this case.